[jit] Shape propagation batch_norm, dropout, quantize, hardswidh

IvanKobzarev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #64751
* #64750
* #64741
* __->__ #64740

Differential Revision: [D30839111](https://our.internmc.facebook.com/intern/diff/D30839111)

Adding shape propagation functions to handle mobilenetv3-large (base and optimized)
(The full change can be seen in https://github.com/pytorch/pytorch/pull/64739 that depends on not landed aot_compiler)

IvanKobzarev committed:
[jit] Shape propagation batch_norm, dropout, quantize, hardswidh

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30839111).

eellison reviewed (commented):
if you rebase, you can just register these as `unary` now after https://github.com/pytorch/pytorch/pull/65095

IvanKobzarev committed:
Update on "[jit] Shape propagation batch_norm, dropout, quantize, hardswidh"


Differential Revision: [D30839111](https://our.internmc.facebook.com/intern/diff/D30839111)

Adding shape propagation functions to handle mobilenetv3-large (base and optimized)
(The full change can be seen in https://github.com/pytorch/pytorch/pull/64739 that depends on not landed aot_compiler)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30839111).

eellison commented on torch/csrc/jit/runtime/symbolic_shape_registry.cpp:
```diff
@@ -436,6 +436,12 @@ const std::string shape_compute_functions =
           assert len(input) == 4
           return conv_output_size(input, weight, bias, stride, padding, dilation, groups)
 
+        def batch_norm(input: List[int], weight: List[int], bias: Optional[List[int]], running_mean: Optional[List[int]], running_var: Optional[List[int]], training: bool, momentum: float, eps: float, cudnn_enabled: bool):
```
Are we not using unary here to distinguish between pointwise/elementwise ops ?

eellison reviewed (approved):
lgtm, question about batch_norm

IvanKobzarev committed:
Update on "[jit] Shape propagation batch_norm, dropout, quantize, hardswidh"


Differential Revision: [D30839111](https://our.internmc.facebook.com/intern/diff/D30839111)

Adding shape propagation functions to handle mobilenetv3-large (base and optimized)
(The full change can be seen in https://github.com/pytorch/pytorch/pull/64739 that depends on not landed aot_compiler)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30839111).
